File: services/api/internal/adapter/mongo/television_documents.go
package mongo

import "time"

const (
	CollPresenters    = "presenters"
	CollProgrammes    = "programmes"
	CollScheduleSlots = "schedule_slots"
)

type presenterDoc struct {
	ID              string  `bson:"_id"`
	Name            string  `bson:"name"`
	Slug            string  `bson:"slug"`
	Locale          string  `bson:"locale"`
	Role            string  `bson:"role"`
	Biography       string  `bson:"biography"`
	PortraitAssetID *string `bson:"portraitAssetId"`
	Published       bool    `bson:"published"`
	CreatedBy       string  `bson:"createdBy"`
}
type programmeDoc struct {
	ID             string   `bson:"_id"`
	Title          string   `bson:"title"`
	Slug           string   `bson:"slug"`
	Locale         string   `bson:"locale"`
	Summary        string   `bson:"summary"`
	Category       string   `bson:"category"`
	PresenterIDs   []string `bson:"presenterIds"`
	ArtworkAssetID *string  `bson:"artworkAssetId"`
	Published      bool     `bson:"published"`
	CreatedBy      string   `bson:"createdBy"`
}
type scheduleSlotDoc struct {
	ID             string    `bson:"_id"`
	ProgrammeID    string    `bson:"programmeId"`
	Locale         string    `bson:"locale"`
	StartsAt       time.Time `bson:"startsAt"`
	EndsAt         time.Time `bson:"endsAt"`
	IsLive         bool      `bson:"isLive"`
	State          string    `bson:"state"`
	ReplayAssetID  *string   `bson:"replayAssetId"`
	CaptionAssetID *string   `bson:"captionAssetId"`
	CreatedBy      string    `bson:"createdBy"`
}

File: packages/adapter-mongo/src/television-documents.ts
import type { ScheduleSlotState } from '@kurasikapa/domain'

export const PRESENTERS = 'presenters'
export const PROGRAMMES = 'programmes'
export const SCHEDULE_SLOTS = 'schedule_slots'

export interface PresenterDocument {
  _id: string
  name: string
  slug: string
  locale: string
  role: string
  biography: string
  portraitAssetId: string | null
  published: boolean
  createdBy: string
}

export interface ProgrammeDocument {
  _id: string
  title: string
  slug: string
  locale: string
  summary: string
  category: string
  presenterIds: string[]
  artworkAssetId: string | null
  published: boolean
  createdBy: string
}

export interface ScheduleSlotDocument {
  _id: string
  programmeId: string
  locale: string
  startsAt: Date
  endsAt: Date
  isLive: boolean
  state: ScheduleSlotState
  replayAssetId: string | null
  captionAssetId: string | null
  createdBy: string
}
